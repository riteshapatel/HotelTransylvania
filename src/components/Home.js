/**
 * @class Home component 
 * @author ritesh.patel
 * App landing screen
 */
import React, {Component} from 'react';
import { Text, View, Image } from 'react-native';
import styles from '../styles/Global.style.js';
import CreepyButton from './CreepyButton'
const logo = require('../assets/logo.png');

class Home extends Component {
    /** navigation options */
    static navigationOptions = {
        title: 'Frankenhomie'
    };

    /** render component */
    render () {
        const {navigate} = this.props.navigation;
        
        return (
            <View style={styles.container}>
                <Image source={logo} />
                <Text style={styles.frankenwelcome}>Welcome to Hotel Transylvania!</Text>
                <Text style={styles.slogan}>
                    Where spooky happens...{'\n'}
                </Text>      
                <CreepyButton title='Creep' navigateTo='Hotels' navigate={navigate}/>     
            </View>
        );
    }
}

export default Home;
