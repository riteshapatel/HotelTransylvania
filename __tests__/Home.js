/**
 * Unit test for Home component 
 * @author ritesh.patel
 */
import React from 'react';
import Home from '../src/components/Home';

import renderer from 'react-test-renderer';

test ('snapshot Home component', () => {
    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(<Home navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
});