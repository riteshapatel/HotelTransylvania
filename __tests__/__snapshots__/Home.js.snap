// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Home component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C800CB",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logo.png",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Hotel Transylvania!
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Where spooky happens...
    

  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#640544",
        "borderColor": "#404040",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginBottom": 5,
        "opacity": 1,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Creep
    </Text>
  </View>
</View>
`;
